### PR TITLE
Address some review comments and refine some wording

### PR DESCRIPTION
**Pull request by Celthi**
Hi @yanqi27 
This PR address the comments you made in https://github.com/yanqi27/core_analyzer/pull/36
and refine some wording.
What's more, it reads `libc version` from debugee.

### CODE REVIEW

File: src/heap_ptmalloc.cpp [hunk 1]
```diff
@@ -2103,26 +2103,24 @@ static bool get_glibc_version(void)
...
 #ifdef FIXME_LIBC_VERSION
 	struct symbol *sym;
 	struct value *val;
 	char* data;
-	/*
-	 * Global var
-	 * File malloc.c: static struct malloc_par mp_;
-	 */
-	sym = lookup_symbol("__libc_version", 0, VAR_DOMAIN, 0).symbol;
+	sym = lookup_symbol("__libc_version", 0, LABEL_DOMAIN, 0).symbol;
```
Comment (Celthi) [inline src/heap_ptmalloc.cpp:2111]: Haven't tested yet.

File: src/heap.cpp [hunk 2]
```diff
@@ -12,19 +12,23 @@
...
 
 
 void register_heap_managers() {
-	#ifndef _WIN32
-	// heap.cpp is used by ref.dll, so we need to only support switching heap in Linux
+	#ifndef WIN32
+	// heap.cpp is used by ref.dll, and we only support switching heap in Linux currently.
 	gCoreAnalyzerHeaps[HeapManagerPtMalloc] = get_pt_malloc_heap_manager();
 	gCoreAnalyzerHeaps[HeapManagerTcMalloc] = get_tc_malloc_heap_manager();
+	gCAHeapers.emplace(std::make_pair<std::string,EnumHeapManager>("pt", HeapManagerPtMalloc));
```
Comment (Celthi) [inline src/heap.cpp:24]: The map is to consolidate all the changes when adding a new heap into a few places, for example, when adding a new heap, we don't need to change the `switch_heap` command.

Comment (Celthi) [reply src/heap.cpp:24]: one question is should we combine the `gCAHeapers` and `gCoreAnalyzerHeaps`. They serve similar purpose.

File: src/heap_ptmalloc.cpp (hunk 1 above)
Comment (Celthi) [reply src/heap_ptmalloc.cpp:2111]: LABEL_DOMAIN does not work either.

Comment (Celthi) [reply src/heap_ptmalloc.cpp:2110]: It is strange that `mp_` can be read with VAR_DOMAIN but __libc_version cannot

Comment (Celthi) [reply src/heap_ptmalloc.cpp:2110]: PS: The `sym` returned by `VAR_DOMAIN` is not null and actually refering to the `__libc_version`, but the real value cannot be got from the returned `sym` either by `valuave_of_variable(sym, 0)` and `value_as_address(val)`

Comment (Celthi) [reply src/heap_ptmalloc.cpp:2110]: I figure out the root cause now. The value is refering to the gdb address space, we need to call  `val = value_coerce_to_target(val);` to cast it to the debugee's address space.

**Comment by Celthi** [comment]
Ready for review~

**Comment by Celthi** [comment]
Hi @yanqi27 With this PR, the `libc_version` can be read from the debuggee now. Could you help to review this PR?

File: src/heap_ptmalloc.cpp [hunk 3]
```diff
@@ -2096,41 +2096,34 @@ static bool traverse_heap_blocks(struct ca_heap* heap,
...
-	int len = strlen(version);
-	int i;
+		strncpy(buf, version, len+1);
 
-	if (len >= bufsz)
-		return false;
+	} else {
+		struct value *val = value_of_variable(sym, 0);
+		if (!read_memory_wrapper(NULL,  value_address(val), buf, TYPE_LENGTH(value_type(val)))) {
+			CA_PRINT("Failed to read thread-local variable \"tcache\"\n");
```
Comment (yanqi27) [inline src/heap_ptmalloc.cpp:2120]: Error message is not the one.

**Review by yanqi27 — COMMENTED** [review]
minor change

File: src/heap_ptmalloc.cpp (hunk 3 above)
Comment (Celthi) [reply src/heap_ptmalloc.cpp]: Done.

**Comment by Celthi** [comment]
Hi @yanqi27 given that I have addressed you review comments and  I want to work on the script mentioned in https://github.com/yanqi27/core_analyzer/commit/72cd45b7e8341cfcd3370007f0a7c8e8bdcef26b#r74206899 today, I'll merge this PR and use your repo as the base branch, I'll merge this PR, so I can work on the script today.

**Pull request merged**